src/commands.js

```diff
@@ -0,0 +1,183 @@
+ */
+function tokenize(command, descriptions){
+    return Q.fcall(function(){
+        var tokens = getTokens(_.toLower(command), descriptions.length);    
+
+        var parameters = {};
+        _.zipWith(tokens, descriptions, function(token, description){
+            if(_.isNil(token)){
+                throw new TooFewTokensError(tokens, descriptions);
+            }else if(description.match(VARIABLE)){
+                parameterizeVariableToken(token, description, parameters);
+            }else if(description.match(CHOICE_VARIABLE)){
+                parameterizeChoiceToken(token, description, parameters);
+            }else if(description.match(CONSTANT_VALUE)){
+                parameterizeConstantToken(token, description, parameters); 
+            }else{
+                throw new InvalidTokenDescriptionError(description);
+            }
+        });
+        return parameters;
+    });
+}
+
+function TooFewTokensError(tokens, descriptions){
+    this.tokens = tokens;
+    this.descriptions = descriptions;
+    this.name = 'TooFewTokensError';
+    this.stack = (new Error()).stack;
+}
+
+function InvalidTokenDescriptionError(description){
+    this.description = description;
+    this.name = 'InvalidTokenDescriptionError';
+    this.stack = (new Error()).stack;
+}
+
+/*
+ * separates the text into numTokens-1, where the remaining words make up the final token
+ * if the number of tokens is greater than the number of " " separted words, the array of 
+ * " " separated words is returned without concatenation
+ */
+function getTokens(text, numTokens){
+    var words = _(text).split(' ').filter().value();
+    if(words.length <= numTokens) return words;
+
+    var tokens = _.slice(words, 0, numTokens-1);
+    var lastToken = _.slice(words, numTokens-1).join(" ");
+    tokens.push(lastToken);
+
+    return tokens;
+}
+
+/*
+ * get the type of the variable from the description
+ * call the specific token verification routine for the type specified
+ * get the key from the description for parameterization
+ * add the variable to the parameter object using the key from the description
+ * if the name or type cannot be determined from the descriptionString, throw InvalidTokenDescription
+ * if the type specified is not known, throw InvalidVariableTypeError
+ */
+function parameterizeVariableToken(token, descriptionString, parameters){
+    //create an object splitting the description string into type and name
+    var description = 
+        _.zipObject(['type', 'name'], _(descriptionString).split(/[\{\:\}]/).filter().value());
+    //if either are not specified, the description string is bad.
+    if(_.isNil(description.type) || _.isNil(description.name)) 
+        throw new InvalidTokenDescriptionError(descriptionString);
+    //for each type defined, test it to see if it matches
+    //   given a match, parse the token and store
+    _.forIn(VARIABLE_TYPES, function (type){
+       if(description.type.match(type.match_expression)){
```

Comment:
i dont think so. just regex happy. you are correct a simple look up would be much more straight forward.